### DISCUSSION TITLE
[NextJS] Next 13 App Directory

### DISCUSSION THREAD

**Discussion by iankduffy** [Sitecore/jss#1498] [Help] [answered]
### Description

Wondering if this library is compatible with the Next 13 App Directory that has just hit stable?

**Comment by ambrauer** [comment, marked as answer]
Yes and no :)

Yes, in that as of JSS [21.1.0](https://github.com/Sitecore/jss/releases/tag/v21.1.0), our Next.js SDK and initializer templates are compatible with Next.js 13.x (and as you mentioned, App Router is stable as of Next.js version 13.4). So, you can introduce the app directory to your JSS application and it should work just fine _alongside the pages directory_. Which leads to ...

No, in that the JSS initializer templates still use the Pages Router by default.

That said, rest assured it is on our radar. The JSS team has been exploring the migration to App Router since beta. Initial POC and discovery revealed some larger roadblocks, but we are actively working with Vercel in order to determine the best path forward for JSS. Stay tuned...

**Reply by m4rvelous252** [reply]
Next 13 server components and the app router is super nice to work with so I'm glad the team is actively working on implementing it 💯 

My team is implementing a headless solution with XMCloud and NextJS and would really love it if there's an App-router starter project

Do you have an estimation of when would such version is released or be in beta ?

**Reply by ambrauer** [reply]
Sorry, no timeline yet. We will try to post an update here when we have one.

**Reply by jeasmith** [reply]
Hi @ambrauer, Have you go an update on this one? Not looking for a timeline, but keen to hear if progress has been made.

**Reply by ambrauer** [reply]
@jeasmith No major update I can provide here, sorry. The JSS team's priorities have shifted a bit, but this is still up there.

**Reply by davidpoczik** [reply]
any update on the topic? or any idea/suggestion how to build our own nextjs(with app router) xmcloud template?

**Reply by FransEngstrom** [reply]
We are also looking at moving over to the app router but for now sitecore jss seems to be the thing in our way, would also love an update.

**Reply by art-alexeyenko** [reply]
Still no concrete timeline for this, folks, sorry. We'll bring you the good news as soon as we're able to.

**Reply by chrissnyder2337** [reply]
Hi @ambrauer, Have any updates on this one? Would be keen to hear if any progress has been made.

**Reply by ambrauer** [reply]
@chrissnyder2337 Unfortunately, no change in status to report on.

**Reply by michaelshowes** [reply]
Annual check-in on this, lol. Is this looking like something viable to happen this year?
